ui/jobs/player.ts

```diff
@@ -0,0 +1,99 @@
+import { EventResponses } from '../../types/event';
+import { Job } from '../../types/job';
+import { NetMatches } from '../../types/net_matches';
+
+import { calcGCDFromStat } from './utils';
+
+export type Stats = Partial<
```

Comment:
Would it be cleaner to not have `Partial` here (and `Required` in all of the functions) and instead have `stats?: Stats` on `Player`?

Reply:
Sounds good, I'll do that!